src/main/webapp/components/TablePaginator.js

```diff
@@ -0,0 +1,102 @@
+import { Component } from "react";
+import { a, hh, ul, li } from 'react-hyperscript-helpers';
+
+export const TablePaginator = hh(class TablePaginator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pageCounter : 0,
+    }
+  }
+
+  goPrevPage = (e) => {
+    const page = this.props.currentPage > 1 ? this.props.currentPage -1 : this.props.currentPage;
+    this.setState( prev => {
+      prev.pageCounter = this.state.pageCounter - 1;
+      return prev;
+    }, () => this.props.onPageChange(page));
+  };
+
+  goNextPage = (e) => {
+    const page = this.props.currentPage < this.props.lastPage ? this.props.currentPage + 1 :this.props.currentPage;
+    this.setState( prev => {
+      prev.pageCounter = this.state.pageCounter + 1;
+      return prev;
+    }, () => this.props.onPageChange(page));
+  };
+
+  goLastPage = (e) => {
+    this.setState( prev => {
+      prev.pageCounter = this.props.lastPage - 1;
+      return prev;
+    }, () => this.props.onPageChange(this.props.lastPage));
+  };
+
+  goFirstPage = (e) => {
+    this.setState( prev => {
+      prev.pageCounter = 0;
+      return prev;
+    }, () => this.props.onPageChange(1));
+  };
+
+  selected = (position) => {
+    this.props.onPageChange(position + this.state.pageCounter);
+    this.setState( prev => {
+      prev.pageCounter = position + this.state.pageCounter - 1;
+      return prev;
+    });
+  };
+
+  render() {
+    return (
+      ul({className: "pagination custom-component"}, [
+        li({ className: "page-item", title: "first page", onClick: (e) => { this.goFirstPage() }}, [
```

Comment:
nit: do you need the `e` if you're not using it?

Reply:
Also a nit: you're using different syntax across these page-item elements, i.e. `onClick: (e) => { this.goFirstPage() }}` vs `onClick: this.goPrevPage`

Reply:
Actually we don't need `e`, so I've just removed them. And also I have changed method's calls to a more consistent syntax.